api/lib/domain/events/handle-badge-acquisition.js

```diff
@@ -13,15 +15,18 @@ const handleBadgeAcquisition = async function({
   checkEventType(event, eventType);
 
   if (completedAssessmentBelongsToACampaign(event)) {
-    const badge = await fetchPossibleCampaignAssociatedBadge(event, badgeRepository);
-    if (isABadgeAssociatedToCampaign(badge)) {
-      const campaignParticipationResult = await fetchCampaignParticipationResults(event, badge, campaignParticipationResultRepository);
-      if (isBadgeAcquired(campaignParticipationResult, badge.badgeCriteria, badgeCriteriaService)) {
-        await badgeAcquisitionRepository.create({
-          badgeId: badge.id,
-          userId: event.userId
-        }, domainTransaction);
-      }
+    const badges = await fetchPossibleCampaignAssociatedBadges(event, badgeRepository);
+    const campaignParticipationResult = await fetchCampaignParticipationResults(event, badges, campaignParticipationResultRepository);
+
+    const badgesBeingAcquired = badges.filter((badge) => isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService));
+    const badgesAcquisitionToCreate = badgesBeingAcquired.map((badge) => {
+      return {
```

Comment:
Je sais que c'était comme ça avant mais je trouve dommage qu'on passe par un objet anonyme ici. 
J'ai l'impression qu'on pourrait en faire un objet et faire `new`, ce qui porterait davantage de sémantique.

Quand un truc est au pluriel j'aime bien mettre le s sur le deuxième terme.

C'est pas non plus le fruit de cette PR mais je trouve `Badge` et `BadgeAcquisition` ambiguës.

Y'en a un qui est "la définition du badge" (`Badge`) et l'autre le faite que "le badge soit acquis" (l'objet anonyme).

Est-ce qu'on pourrait pas appeler l'objet anonyme `AcquiredBadge` ?

Je radote mais si du coup `Badge` correspond à "la définition du badge", je pense que `is(Badge)Acquired()` devrait en être une de ses méthodes. Je comprend pas pourquoi on passer par un service ici. Si jamais ça reste confusant on pourrait renommer `Badge` en `BadgeDefinition` (ou autre chose du style, qui serait peut-être pas super beau mais qui aurait le mérite de clarifier la dualité).